cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudsqlMysqlAction.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.cloudsql.mysql;
+
+import com.google.common.collect.ImmutableMap;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.plugin.db.batch.action.AbstractDBAction;
+import io.cdap.plugin.db.batch.action.QueryConfig;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Action that runs a MySQL command on a CloudSQL instance database. */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(CloudsqlMysqlConstants.PLUGIN_NAME)
+@Description("Action that runs a MySQL command on a CloudSQL instance database")
+public class CloudsqlMysqlAction extends AbstractDBAction {
+
+  private final CloudsqlMysqlActionConfig cloudsqlMysqlActionConfig;
+
+  public CloudsqlMysqlAction(CloudsqlMysqlActionConfig cloudsqlMysqlActionConfig) {
+    super(cloudsqlMysqlActionConfig, false);
+    this.cloudsqlMysqlActionConfig = cloudsqlMysqlActionConfig;
+  }
+
+  /** Mysql Action Config. */
+  public static class CloudsqlMysqlActionConfig extends QueryConfig {
+  
+    @Name(CloudsqlMysqlConstants.CONNECTION_NAME)
+    @Description(
+        "The CloudSQL instance to connect to. For a public instance, the connection string should be in the format "
+            + "<PROJECT_ID>:<REGION>:<INSTANCE_NAME> which can be found in the instance overview page. For a private "
+            + "instance, enter the internal IP address of the Compute Engine VM cloudsql proxy is running on.")
+    public String connectionName;
+
+    @Name(DATABASE)
+    @Description("Database name to connect to")
+    public String database;
+
+    @Name(CloudsqlMysqlConstants.CONNECTION_TIMEOUT)
+    @Description(
+        "The timeout value used for socket connect operations. If connecting to the server takes longer"
+            + " than this value, the connection is broken. "
```

Comment:
Should be consistent with line breaks. A portion of the next line can be on this line.